DOI.js

```diff
@@ -93,33 +95,47 @@ function completeDOIs(doc) {
 	}
 }
 
-function retrieveDOIs(DOIs, doc) {
-	__num_DOIs = DOIs.length;
+function retrieveDOIs(dois, doc, providers) {
+	var numDois = dois.length;
+	var provider = providers.shift();
+	
+	var remainingDOIs = dois.slice();//copy array but not by reference
 
-	for(var i=0, n=DOIs.length; i<n; i++) {
+	for(var i=0, n=dois.length; i<n; i++) {
 		(function(doc, DOI) {
 			var translate = Zotero.loadTranslator("search");
-			translate.setTranslator("11645bd1-0420-45c1-badb-53fb41eeb753");
+			translate.setTranslator(provider.id);
 	
 			var item = {"itemType":"journalArticle", "DOI":DOI};
 			translate.setSearch(item);
 	
 			// don't save when item is done
 			translate.setHandler("itemDone", function(translate, item) {
+				selectArray[item.DOI] = item.title;
 				if (!item.title) {
-					Zotero.debug("No title available for " + DOI);
-					return;
+					Zotero.debug("No title available for " + item.DOI);
+					item.title = "[No Title]";
+					selectArray[item.DOI] = "[" + item.DOI + "]";
+				}
+				items[item.DOI] = item;
+
+				// done means not remaining anymore
+				if (remainingDOIs.indexOf(item.DOI) > -1) {
+					remainingDOIs.splice(remainingDOIs.indexOf(item.DOI), 1);
+				} else {
+					Z.debug(item.DOI + " not anymore in the list of remainingDOIs = " + remainingDOIs);
```

Comment:
Yeah, always surprised by how unforgiving `print(..)` is in Python. Yet another reason to use format expressions.